src/aui/auibook.cpp

```diff
@@ -3446,6 +3446,140 @@ bool wxAuiNotebook::InsertPage(size_t index, wxWindow *page,
     }
 }
 
+// Helper class to calculate the best size of a wxAuiNotebook
+class wxAuiLayoutObject
+{
+public:
+    wxAuiLayoutObject(const wxSize &size,
+                      const wxAuiPaneInfo &pInfo) : m_pInfo(pInfo)
+    {
+        m_size = size;
+        /* To speed up the sorting of the panes, the direction is mapped to a
+         * useful increasing value. This avoids complicated comparison of the
+         * enum values during the sort. The size calculation is done from the
+         * inner to the outermost direction. Therefore CENTER < LEFT/RIGHT <
+         * TOP/BOTTOM (It doesn't matter it LEFT or RIGHT is done first, as
+         * both extend the best size horizontally; the same applies for
+         * TOP/BOTTOM in vertical direction) */
+        switch (pInfo.dock_direction)
+        {
+            case wxAUI_DOCK_CENTER: m_dir = 0; break;
+            case wxAUI_DOCK_LEFT:   m_dir = 1; break;
+            case wxAUI_DOCK_RIGHT:  m_dir = 2; break;
+            case wxAUI_DOCK_TOP:    m_dir = 3; break;
+            case wxAUI_DOCK_BOTTOM: m_dir = 4; break;
+            default:                m_dir = 5;
+        }
+    }
+    void MergeLayout(const wxAuiLayoutObject &lo2)
+    {
+        if (this == &lo2)
+            return;
+
+        bool mergeHorizontal;
+        if (m_pInfo.dock_layer != lo2.m_pInfo.dock_layer || m_dir != lo2.m_dir)
+            mergeHorizontal = lo2.m_dir < 3;
+        else if (m_pInfo.dock_row != lo2.m_pInfo.dock_row)
+            mergeHorizontal = true;
+        else
+            mergeHorizontal = lo2.m_dir >= 3;
+
+        if (mergeHorizontal)
+        {
+            m_size.x += lo2.m_size.x;
+            if (lo2.m_size.y > m_size.y) m_size.y = lo2.m_size.y;
+        }
+        else
+        {
+            if (lo2.m_size.x > m_size.x) m_size.x = lo2.m_size.x;
+            m_size.y += lo2.m_size.y;
+        }
+    }
+
+    wxSize m_size;
+    const wxAuiPaneInfo &m_pInfo;
```

Comment:
As Travis CI failure shows, we can't use a const reference here as this class becomes neither copyable nor movable then. We can either make a copy (i.e. drop the reference part) or use a pointer, which is uglier but avoids copying. Or revert back to using individual `m_layer`, `m_dir` and `m_row` fields here, while still using `wxAuiPaneInfo` to initialize them in the ctor.

I'm leaving the choice to you, but we do need to fix the build.

Reply:
Of course the build must work. I corrected the reference.